new-packages/ts-project/__tests__/actions.test.ts

```diff
@@ -0,0 +1,691 @@
+    .toMatchInlineSnapshot(`
+      [
+        [
+          {
+            "blocks": {
+              "block-0": {
+                "blockType": "action",
+                "parentId": "state-0",
+                "properties": {
+                  "params": {},
+                  "type": "callAnders",
+                },
+                "sourceId": "callAnders",
+                "uniqueId": "block-0",
+              },
+            },
+            "edges": {},
+            "implementations": {
+              "actions": {
+                "undefined": {
+                  "id": "callAnders",
+                  "name": "callAnders",
+                  "type": "action",
+                },
+              },
+              "actors": {},
+              "guards": {},
+            },
+            "nodes": {
+              "state-0": {
+                "data": {
+                  "description": undefined,
+                  "entry": [
+                    "block-0",
+                  ],
+                  "exit": [],
+                  "history": undefined,
+                  "initial": undefined,
+                  "invoke": [],
+                  "metaEntries": [],
+                  "tags": [],
+                  "type": "normal",
+                },
+                "parentId": undefined,
+                "type": "node",
+                "uniqueId": "state-0",
+              },
+            },
+            "root": "state-0",
+          },
+          [],
+        ],
+      ]
+    `);
+});
+
+test('should extract multiple string actions', async () => {
+  const tmpPath = await testdir({
+    'tsconfig.json': JSON.stringify({}),
+    'index.ts': ts`
+      import { createMachine } from "xstate";
+
+      createMachine({
+        entry: ["callDavid", "callAnders"],
+      });
+    `,
+  });
+
+  const project = await createTestProject(tmpPath);
+  expect(replaceUniqueIds(project.extractMachines('index.ts')))
+    .toMatchInlineSnapshot(`
+      [
+        [
+          {
+            "blocks": {
+              "block-0": {
+                "blockType": "action",
+                "parentId": "state-0",
+                "properties": {
+                  "params": {},
+                  "type": "callDavid",
+                },
+                "sourceId": "callDavid",
+                "uniqueId": "block-0",
+              },
+              "block-1": {
+                "blockType": "action",
+                "parentId": "state-0",
+                "properties": {
+                  "params": {},
+                  "type": "callAnders",
+                },
+                "sourceId": "callAnders",
+                "uniqueId": "block-1",
+              },
+            },
+            "edges": {},
+            "implementations": {
+              "actions": {
+                "undefined": {
+                  "id": "callAnders",
+                  "name": "callAnders",
+                  "type": "action",
+                },
```

Comment:
We should see two actions here because we have two action blocks with different source IDs.

Reply:
Fixed - they got "merged" because both were keyed under `"undefined"` so the last one won this fight.